### PR TITLE
[Swift] Add Kitura-NIO

### PR DESCRIPTION
**Pull request by shial4**
Before submitting your PR, please review the following checklist :

## If you are adding a framework

+ [x] Please check configuration files
   + [x] `neph.yml`, for [neph](http://tbrand.github.io/neph) job processing
   + [x] `FRAMEWORKS.yml`, for **frameworks** list
   + [x] `travis.yml`, for CI
+ [x] A `Dockerfile` exists ?
+ [x] All tests passes ?
~~~
export FRAMEWORK=<MY_FRAMEWORK>
shards install
shards build --release
bin/neph ${FRAMEWORK}
crystal spec
~~~

### CODE REVIEW

File: swift/kitura-nio/main.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import Kitura
+
+let router = Router()
+
+router.get("/") { _, res, next in
+    try res.send("")
+    next()
+}
+
+router.get("/user/:id") {  req, res, next in
```
Comment (waghanza) [inline swift/kitura-nio/main.swift]: can this be constrained as digit ?

**Comment by waghanza** [comment]
@shial4 can you update also `kitura` ?

**Comment by shial4** [comment]
Updated @waghanza  :) 
Can't wait to compare the results!

File: swift/kitura-nio/main.swift (hunk 1 above)
Comment (shial4) [reply swift/kitura-nio/main.swift]: Not sure If I do understand, what exactly do you want to achieve?

Comment (shial4) [reply swift/kitura-nio/main.swift]: We could use codable route. For instance:
```
//contains an identifier of type `Int`
func getUserByIdHandler(id: Int, completion: (Int, RequestError?) -> Void) {
    completion(id, nil)
}

router.get("/user/:id", handler: app.getUserByIdHandler)
```

Or we could use raw request and manually convert it to Int.

Comment (waghanza) [reply swift/kitura-nio/main.swift]: Up to you :stuck_out_tongue: 

I honestly prefer using what is considered as a `best-practice` in `swift (clean / re-usable code)

Comment (shial4) [reply swift/kitura-nio/main.swift]: Either are clean. Codable is model convertible language stuff. I do like raw requests style.

**Comment by waghanza** [comment]
Close #1279

File: swift/kitura-nio/main.swift (hunk 1 above)
Comment (waghanza) [reply swift/kitura-nio/main.swift]: Could you homogenize also for `kitura` ?

I will update result **tomorrow**

**Comment by shial4** [comment]
@waghanza  Specifies that only digits should be matched for user/:id route.
In both Kitura & Kitura-NIO example.

**Pull request merged**